bitcoin/src/network/constants.rs

```diff
@@ -92,22 +90,115 @@ impl Network {
     /// # Examples
     ///
     /// ```rust
-    /// use bitcoin::network::constants::Network;
+    /// use bitcoin::network::constants::{Network, Magic};
     ///
     /// let network = Network::Bitcoin;
-    /// assert_eq!(network.magic(), 0xD9B4BEF9);
+    /// assert_eq!(network.magic(), Magic::from_bytes([0xF9, 0xBE, 0xB4, 0xD9]));
     /// ```
-    pub fn magic(self) -> u32 {
-        // Note: any new entries here must be added to `from_magic` above
+    pub fn magic(self) -> Magic {
+        Magic::from(self)
+    }
+}
+
+/// Network magic bytes to identify the cryptocurrency network the message was intended for.
+#[derive(Copy, PartialEq, Eq, PartialOrd, Ord, Clone, Hash)]
+pub struct Magic([u8; 4]);
+
+impl Magic {
+    /// Create network magic from bytes.
+    pub fn from_bytes(bytes: [u8; 4]) -> Magic {
```

Comment:
This implies endianness, perhaps we should have `from_be_bytes` and `from_le_bytes` (and `to_be_bytes`, `to_le_bytes`)

Reply:
I don't think it does, we remove endianess issues completely with this PR by having everything in the network order. Maybe `from_network_bytes` if we want to be extra clear would work but doesn't seem necessary. A doc comment would be probably useful though.

Reply:
I don't understand what endianness has to do with this type? These are four bytes that are put onto the wire and are never interpreted as a number.

Reply:
My mistake, sorry for the noise.